docs/porting/target/static_pinmap.md

```diff
@@ -0,0 +1,180 @@
+   #define PINMAP_CAN_RD [PinMap CAN RD]
+   #define PINMAP_CAN_TD [PinMap CAN RD]
+   ```
+
+- Provide `STATIC_PINMAP_READY` macro in `PinNames.h`  
+
+   Adding this macro enables the static pin map support for the target.
+
+   ```
+   /* If this macro is defined, you can use constexpr utility functions for pin map search. */
+   #define STATIC_PINMAP_READY 1
+   ```
+
+## Testing
+
+Use the code below to test the static pin map extension:
+
+```
+int main()
+{
+    /* Regular use */
+    SPI spi(D1, D2, D3, D4);
+
+    /* Static pinmap */
+    const spi_pinmap_t static_spi_pinmap = {SPI_1, D1, 2, D2, 2, D3, 2, D4, 2};
+    SPI spi(static_spi_pinmap);
+
+    /* Static pinmap with constexpr utility function */
+    constexpr spi_pinmap_t static_spi_pinmap = get_spi_pinmap(D1, D2, D3, D4);
+    SPI spi(static_spi_pinmap);
+
+    return 0;
+}
+```
+
+When you use the static pin map extension, you save on ROM:  
+
+```
+| Module                          |        .text |   .data |       .bss |
+|---------------------------------|--------------|---------|------------|
+| [lib]\c_w.l                     |    11175(+0) |  16(+0) |    348(+0) |
+| [lib]\fz_wm.l                   |       34(+0) |   0(+0) |      0(+0) |
+| [lib]\m_wm.l                    |       48(+0) |   0(+0) |      0(+0) |
+| anon$$obj.o                     |       32(+0) |   0(+0) | 197888(+0) |
+| drivers\source                  |      192(+0) |   0(+0) |      0(+0) |
+| features\netsocket              |      143(+0) |   0(+0) |      0(+0) |
+| hal\mbed_critical_section_api.o |      154(+0) |   0(+0) |      2(+0) |
+| hal\mbed_gpio.o                 |       96(+0) |   0(+0) |      0(+0) |
+| hal\mbed_pinmap_common.o        |      0(-272) |   0(+0) |      0(+0) | // remove pinmap lib (this is common for all peripherals)
+| hal\mbed_ticker_api.o           |      978(+0) |   0(+0) |      0(+0) |
+| hal\mbed_us_ticker_api.o        |      114(+0) |   4(+0) |     65(+0) |
+| main.o                          |      70(+32) |   0(+0) |      0(+0) | // extra space for static pin map structure in application
+| platform\source                 |    5683(+46) |  64(+0) |    249(+0) | // extra space for UART static pin map structure to initialize the console
+| rtos\source                     |     8990(+0) | 168(+0) |   6626(+0) |
+| targets\TARGET_Freescale        |  16581(-816) |  12(+0) |    340(+0) | // remove pin maps and driver code reduction
+| Subtotals                       | 44290(-1010) | 264(+0) | 205518(+0) |
+Total static RAM memory (data + bss): 205782(+0) bytes
+Total flash memory (text + data): 44554(-1010) bytes
+```
+
+Run FPGA tests to check whether your implementation is valid:
+
+```
+ mbed test -t ARM -m K64F -n tests-mbed_hal_fpga_ci_test_shield*
+```
+
+<span class="notes">**Note:** Your target must be ready to run FPGA-Test-Shield tests. You can test the following peripherals: `Analogin`, `SPI`, `I2C`, `PWM`, `UART`.</span>
```

Comment:
Is this list likely to change? If so and it's easy for it to become out of date, I suggest removing it.

Reply:
I will remove the list.